Automattic/sensei
# Display all lessons and modules in preview mode in the Outline block

Fixes #3691

Depends on #3697

### Changes proposed in this Pull Request

* Show all lessons and modules in preview mode (if the user has edit access)
* Cache block HTML so it's only rendered once
  * Sensei unsupported theme handler would re-render the content, including this block — with a different WP_Query, missing the preview status

### Testing instructions

* Edit a course with outline and have draft lessons or empty modules
* Preview the course in a new tab
* All lessons and modules show be visible, with a notice that some are drafts
--
* Only published content should be visible in non-preview mode or to non-admin users
--
* Publish all lessons, remove empty modules
* Notice should not be displayed in preview mode


### Screenshot / Video

<img width="711" alt="image" src="https://user-images.githubusercontent.com/176949/95789035-0e2bf680-0cdd-11eb-8136-193a1aac6b6a.png">
